packages/Modal/src/Modal.js

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import extractChildren from "@paprika/helpers/lib/extractChildren";
+import * as styled from "./Modal.styles";
+
+const propTypes = {
+  /** The content for the Modal. */
+  children: PropTypes.node.isRequired,
+
+  /** Control the visibility of the modal */
+  isOpen: PropTypes.bool.isRequired,
+
+  /** Callback triggered when the modal needs to be close */
+  onClose: PropTypes.func,
+
+  /** Callback once the modal has been opened event */
+  onAfterOpen: PropTypes.func,
+
+  /** Callback once the modal has been closed event */
+  onAfterClose: PropTypes.func,
+
+  /* Control the size (max-width) of the modal */
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  onAfterClose: () => {},
+  onClose: () => {},
+  onAfterOpen: () => {},
+  size: ShirtSizes.MEDIUM,
+};
+
+const Modal = props => {
+  const { isOpen, onClose, onAfterClose, onAfterOpen, size, ...moreProps } = props;
+
+  const {
+    "Modal.FocusTrap": focusTrapExtracted,
+    "Modal.Header": headerExtracted,
+    "Modal.Content": contentExtracted,
+    "Modal.Footer": footerExtracted,
+    children,
+  } = extractChildren(moreProps.children, ["Modal.FocusTrap", "Modal.Header", "Modal.Content", "Modal.Footer"]);
+
+  const extendedFocusTrapOptions = focusTrapExtracted ? focusTrapExtracted.props : {};
+
+  return (
+    <styled.Overlay
+      isOpen={isOpen}
+      onClose={onClose}
+      onAfterOpen={onAfterOpen}
+      onAfterClose={onAfterClose}
+      focusTrapOptions={extendedFocusTrapOptions}
+    >
+      {state => (
+        <styled.Wrapper size={size}>
+          <styled.Dialog state={state} role="dialog" data-pka-anchor="modal">
```

Comment:
The element with `role=dialog` should also have `aria-modal=true` and either `aria-label` or `aria-labelledby`.  While the preferred approach is probably to use `aria-labelledby` with the `id` for the `<Heading>` element, that's a bit awkward with this composition, and I think it's probably easier for consumer to simply put the same content into the `aria-label` (but lets call the prop `a11yText`, to maintain our convention, and to insulate typical consumer from the implementation details).

Reference:
https://www.w3.org/TR/wai-aria-practices/#dialog_roles_states_props